src/client/actions.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Actions are the mechanism through which notifications send messages back to the application that raised them. The
+ * service defines a number of ways in which actions can be raised (typically user interactions, such as clicking a
+ * button), and it is up to each application to decide if it wishes to be informed of these interactions.
+ *
+ * For an application to receive an event on each of these actions, the application should specify the message it
+ * wishes to receive when that event happens. This is known as the action {@link NotificationActionResult|result}.
+ *
+ * All actions passed back to an application come through a single entry point. The application should attach a
+ * listener for the {@link NotificationActionEvent|`notification-action`} event. This listener will be called once for
+ * each action raised by the service.
+ *
+ * For an overview of actions, consider the sample notification below:
+ * ```ts
+ * import {addEventListener, create} from 'openfin-notifications';
+ *
+ * // Create a notification with two buttons
+ * create({
+ *     // Basic info
+ *     title: 'Reminder',
+ *     body: 'Event "Weekly Meeting" is starting soon...',
+ *     category: 'Upcoming Events',
+ *
+ *     // We'll use the 'customData' field to store metadata about the event
+ *     customData: {eventId: '12345'},
+ *
+ *     // We want the user clicking the notification to open the associated event, so register an 'onSelect' action
+ *     onSelect: {action: 'view-event', target: 'popup'},
+ *
+ *     buttons: [
+ *         // A button that closes the notification and doesn't prompt the user about this event again. Since there is
+ *         // no special action required to handle clicking this button, there is no 'onClick' payload. This means that
+ *         // the application will not receive a "notification-action" event when the button is clicked
+ *         {
+ *             title: 'Dismiss',
+ *             iconUrl: 'https://www.example.com/cancel.png'
+ *         },
+ *
+ *         // A button that will schedule another reminder for 5 minutes from now. Since the application will need to
+ *         // perform an action on this interaction, it will need to know about the user clicking this button. By
```

Comment:
We've given a very specific definition of action at the top of this file, but this isn't the kind of "action" we mean here.

"Since the application will be responsible for snoozing the event, it will need to know about the user clicking this button"?

Reply:
Also feels odd that we're going into our full explanation on the *second* use of an ActionResult

Reply:
> We've given a very specific definition of action at the top of this file, but this isn't the kind of "action" we mean here.
> 
> "Since the application will be responsible for snoozing the event, it will need to know about the user clicking this button"?

_Kiiiinda_ is... Part of the reason for calling these things "actions" is because the result payload should represent a specific action that you want the app to take - for example, scheduling a reminder. But yeah, will edit - think the way it's worded here more muddles than clarifies.

> Also feels odd that we're going into our full explanation on the second use of an ActionResult

As in, this duplicates lines 6-7 above? This went through a few iterations - may well make sense to trim-down the description here a little.

Wasn't sure if maybe the description above was a bit "detached", and would make more sense in this kind of a context - with the button definition right below the description.